src/main/java/com/sem/btrouble/controller/CollisionHandler.java

```diff
@@ -176,6 +182,37 @@ private void onCollide(Shape collider, Shape collidee) {
         if (collider instanceof Wall) {
             wallCollide((Wall) collider, collidee);
         }
+        
+        if (collider instanceof PowerUp) {
+        	powerCollide((PowerUp) collider, collidee);
+        }
+    }
+    
+    private void powerCollide(PowerUp power, Shape collidee) {
+    	if (collidee instanceof Floor) {
+    		power.setFalling(false);
+    		power.setY(collidee.getY() - power.getHeight());
+    	}
+    	
+        if (collidee instanceof Player) {
+        	ArrayList<Player> players = Model.getPlayers();
+        	Player player = null;
+        	for(int i = 0; i < players.size(); i++) {
```

Comment:
Je zou hier ook met een foreach kunnen werken:
for(Player otherplayer: players) {
    if(collidee.equals(otherplayer)) {
        player = (Player) collidee;
    }
}